addon/usables/use-machine.ts

```diff
@@ -126,19 +133,53 @@ export class Statechart<
    * Private
    */
 
-  private get [CONFIG]() {
-    return this.args.named;
-  }
-
   @action
   send(...args: SendArgs<Context, Schema, Event>) {
     return this[INTERPRETER].send(...args);
   }
 
   @action
-  private _setupMachine() {
-    this.withContext(this[CONFIG]?.context);
-    this.withConfig(this[CONFIG]?.config);
+  private _setupMachine() {}
+
+  /**
+   * Lifecycle methods on Resource
+   *
+   */
+  @action
+  protected setup() {
+    console.log('this.setup');
+    let { chart, context, config, owner } = this.args.positional?.[0];
+
+    assert(ERROR_CHART_MISSING, chart);
+
+    let gatherActions = function (object, actionsArray) {
```

Comment:
can this be extracted outside the class?
it doesn't need the `this` and currently displays implementation details that aren't super necessary for understanding how `setup` works